File: src/main/kotlin/br/com/lgrapplication/bdr/prova/model/Case.java
package br.com.lgrapplication.bdr.prova.model;

import br.com.lgrapplication.bdr.prova.model.enums.Access;
import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
import org.springframework.data.annotation.Id;
import javax.validation.constraints.Max;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
public class Case implements Serializable {

    private static final long serialVersionUID = 7526472295622776147L;

    @Id
    private String id;
    @Max(40)
    private String folder;
    @NotNull
    private String client;

    @NotNull
    private String title;
    private List<String> tags;
    private String description;
    private String comments;
    private String responsible;
    private Access access;
    @NotNull
    private LocalDateTime dtCreated = LocalDateTime.now();

    public Case(){
        this.id = UUID.randomUUID().toString();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getFolder() {
        return folder;
    }

    public void setFolder(String folder) {
        this.folder = folder;
    }

    public String getClient() {
        return client;
    }

    public void setClient(String client) {
        this.client = client;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<String> getTags() {
        return tags;
    }

    public void setTags(List<String> tags) {
        this.tags = tags;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getComments() {
        return comments;
    }

    public void setComments(String comments) {
        this.comments = comments;
    }

    public String getResponsible() {
        return responsible;
    }

    public void setResponsible(String responsible) {
        this.responsible = responsible;
    }

    public Access getAccess() {
        return access;
    }

    public void setAccess(Access access) {
        this.access = access;
    }

    public LocalDateTime getDtCreated() {
        return dtCreated;
    }

    public void setDtCreated(LocalDateTime dtCreated) {
        this.dtCreated = dtCreated;
    }
}

File: src/main/kotlin/br/com/lgrapplication/bdr/prova/prova/model/Case.kt
package br.com.lgrapplication.bdr.prova.prova.model

import br.com.lgrapplication.bdr.prova.prova.model.enum.Access
import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity
import org.springframework.data.annotation.Id
import java.time.LocalDateTime
import java.util.*
import javax.annotation.Generated
import javax.validation.constraints.Max
import javax.validation.constraints.NotNull
import javax.xml.bind.annotation.XmlEnumValue

@Entity
data class Case (
        @Id
        @Generated()
        var id: String = UUID.randomUUID().toString(),
        @Max(40)
        val folder : String,
        @NotNull
        val client : String,
        @NotNull
        val title : String,
        val tags: List<String>,
        val description: String,
        val comments: String,
        val responsible: String,
        val access: Access,
        @NotNull
        val dtCreated : LocalDateTime = LocalDateTime.now()) {
        //constructor() {
                //()
                //@Enumerated(EnumType.STRING)
        //}
}
